mypyc/irbuild/function.py

```diff
@@ -816,15 +833,25 @@ def generate_singledispatch_dispatch_function(
     current_func_decl = builder.mapper.func_to_decl[fitem]
     arg_info = get_args(builder, current_func_decl.sig.args, line)
 
-    def gen_func_call_and_return(func_name: str) -> None:
-        func = builder.load_global_str(func_name, line)
-        # TODO: don't pass optional arguments if they weren't passed to this function
+    def gen_func_call_and_return(func_name: str, fullname: Optional[str] = None) -> None:
+        func = load_func(builder, func_name, fullname, line)
         ret_val = builder.builder.py_call(
             func, arg_info.args, line, arg_info.arg_kinds, arg_info.arg_names
         )
         coerced = builder.coerce(ret_val, current_func_decl.sig.ret_type, line)
         builder.nonlocal_control[-1].gen_return(builder, coerced, line)
 
+    # Add all necessary imports of other modules that have registered functions in other modules
```

Comment:
I think that this is fine for now, but once we properly support doing native calls, we will want to skip adding the imports for those.

Reply:
Add a note to that effect, maybe

Reply:
I added a comment with a note about this.